examples/material_gallery/lib/demo/data_table_demo.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/material.dart';
+
+class Desert {
+  Desert(this.name, this.calories, this.fat, this.carbs, this.protein, this.sodium, this.calcium, this.iron);
+  final String name;
+  final int calories;
+  final double fat;
+  final int carbs;
+  final double protein;
+  final int sodium;
+  final int calcium;
+  final int iron;
+
+  bool selected = false;
+}
+
+class DataTableDemo extends StatefulWidget {
+  @override
+  _DataTableDemoState createState() => new _DataTableDemoState();
+}
+
+class _DataTableDemoState extends State<DataTableDemo> {
+
+  int _sortColumn;
+  bool _sortAscending = true;
+
+  final List<Desert> _deserts = [
+    new Desert('Frozen yogurt',      159,  6.0, 24, 4.0,  87, 14,  1),
+    new Desert('Ice cream sandwich', 237,  9.0, 37, 4.3, 129,  8,  1),
+    new Desert('Eclair',             262, 16.0, 24, 6.0, 337,  6,  7),
+    new Desert('Cupcake',            305,  3.7, 67, 4.3, 413,  3,  8),
+    new Desert('Gingerbread',        356, 16.0, 49, 3.9, 327,  7, 16),
+    new Desert('Jelly bean',         375,  0.0, 94, 0.0,  50,  0,  0),
+    new Desert('Lollipop',           392,  0.2, 98, 0.0,  38,  0,  2),
+    new Desert('Honeycomb',          408,  3.2, 87, 6.5, 562,  0, 45),
+    new Desert('Donut',              452, 25.0, 51, 4.9, 326,  2, 22),
+    new Desert('KitKat',             518, 26.0, 65, 7.0,  54, 12,  6),
+  ];
+
+  void _sort/*<T>*/(Comparable<dynamic/*=T*/> getField(Desert d), int columnIndex, bool ascending) {
```

Comment:
I hope that the support for these type parameters arrives before we deliver this app to the general public.